subsys/net/lib/nrf_cloud/src/nrf_cloud_alerts.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <net/nrf_cloud.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <date_time.h>
+#include "nrf_cloud_fsm.h"
+#include "nrf_cloud_mem.h"
+#include "nrf_cloud_codec.h"
+#include "nrf_cloud_transport.h"
+#include <net/nrf_cloud_rest.h>
+#include <net/nrf_cloud_alerts.h>
+
+LOG_MODULE_REGISTER(nrf_cloud_alerts, CONFIG_NRF_CLOUD_LOG_LEVEL);
+
+static int alerts_enabled = IS_ENABLED(CONFIG_NRF_CLOUD_ALERTS);
+
+#if defined(CONFIG_NRF_CLOUD_ALERTS)
+static atomic_t alert_sequence;
+
+static int alert_prepare(struct nrf_cloud_data *output,
+			 enum nrf_cloud_alert_type type,
+			 float value,
+			 const char *description)
+{
+	struct nrf_cloud_alert_info alert = {.ts_ms = 0, .type = type, .value = value,
+					     .description = description};
+	int64_t timestamp;
+
+	if (IS_ENABLED(CONFIG_DATE_TIME)) {
+		if (!date_time_now(&timestamp)) {
+			alert.ts_ms = timestamp;
+		}
+	}
+
+	if (!alert.ts_ms || IS_ENABLED(CONFIG_NRF_CLOUD_ALERTS_SEQ_ALWAYS)) {
+		alert.sequence = alert_sequence++;
```

Comment:
I'm not sure it matters.  Alerts sent before NTP or modem time are available to set the clock will use the sequence due to ts_ms == 0.

Once NTP or modem time are available at least one time, the runtime library's clock is set and runs on its own after that.  So it should always have a timestamp after that first initial set.

IOW, I don't think it's possible for date_time_now() to return an error at all after the first time it returns no error (because the clock is set).

Reply:
i meant if the sequence number is not included in the message, should it still be incremented?
i couldn't think of a good reason why it would matter either way.

another thing that probably doesn't matter 😅  
is it intentional to send zero as the first sequence number?